lib/vf_settokenvalue.cpp

```diff
@@ -392,6 +392,7 @@ namespace ValueFlow
             // is condition always true/false?
             if (parent->astOperand1()->hasKnownValue()) {
                 const Value &condvalue = parent->astOperand1()->values().front();
+                assert(condvalue.isKnown());
```

Comment:
This fails in at least `TestLeakAutoVar::assign26`. It has two values - the first is `INT` and `Possible`, and the other is `BUFFER_SIZE` and `Known`.

Reply:
@pfultz2 Should this be actually checking for a known `INT` value?

Reply:
@pfultz2 ping

Reply:
Well its checking the condition based on a known int value or known tok value. I dont know if the known tok value is needed anymore.

Reply:
Oh - I didn't look at the subsequent code. Sorry about that.

Based on that I think it should be simply calling `getKnownValue()`. Will give it a spin tomorrow.